clients/drcachesim/scheduler/scheduler.cpp

```diff
@@ -928,7 +940,7 @@ scheduler_tmpl_t<RecordType, ReaderType>::read_traced_schedule()
         // a signal can come in after a prior timestamp with no intervening instrs).
         if (!all_sched[cur_output].empty() &&
             input == all_sched[cur_output].back().key.input &&
-            start == all_sched[cur_output].back().start_instruction) {
+            start == all_sched[cur_output].back().value.start_instruction) {
```

Comment:
Do we need to confirm whether all_sched[cur_output].back() is not an IDLE entry? Same elsewhere.

Reply:
Here it cannot be, by construction: this is an as-traced schedule from which we construct all_sched from only DEFAULT types.  I added an assert here as a sanity check and another one in the check_and_fix_* routine.